packages/iov-bns-governance/src/proposals.ts

```diff
@@ -0,0 +1,90 @@
+import { ReadonlyDate } from "readonly-date";
+import { As } from "type-tagger";
+
+import { Address, Amount } from "@iov/bcp";
+import { Fraction } from "@iov/bns";
+
+export type CommitteeId = number & As<"committee">;
+
+export enum ProposalType {
+  AddCommitteeMember,
+  RemoveCommitteeMember,
+  AmendCommitteeThreshold,
+  AmendCommitteeQuorum,
+  AddValidator,
+  RemoveValidator,
+  AccessGuaranteeFund,
+  DistributeFunds,
+  AmendProtocol,
+}
+
+export interface CommonProposalOptions {
+  readonly type: ProposalType;
+  readonly title?: string;
```

Comment:
Every proposal needs a title and a description (weave says "Human readable title. Must match `^[a-zA-Z0-9 _.-]{4,128}$`"), so this cannot be optional.

Reply:
So currently a default is provided in the transaction builder. I was pretty unclear on the exact roles of `title` vs `description` vs `text`, so I thought we might want a fairly generic/uniform title by default (with the option to override) to avoid a mess?